example/subelements/basic_quad_refinement.cxx

```diff
@@ -0,0 +1,125 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <sc_refcount.h>
+#include <sc_shmem.h>
+#include <p4est_connectivity.h>
+#include <p8est_connectivity.h>
+#include <t8_schemes/t8_new_feature/t8_subelements_cxx.hxx>
+#include <t8_forest/t8_forest_adapt.h>
+#include <t8_forest.h>
+#include <t8_cmesh_vtk.h>
+#include <t8_vec.h>
+#include <example/common/t8_example_common.h>
+
```

Comment:
Please add a comment that explains what this example does.

Also comment for each function at least briefly what it does.

Reply:
done